### ISSUE TITLE
[Accessibility][Navigation] Hamburger Menu

### ISSUE THREAD

**Issue by zajcu** [adobe/aem-core-wcm-components#680]
## Feature Request in Navigation Component

The standard is to have the possibility to render navigation component as collapsed hamburger menu on mobile devices. 

**Describe the solution you'd like**
The component dialog should be extended to have a checkbox for example Collapse Mode. When the checkbox is selected the textfield is displaying, this text will be used as a aria label.
Additionally when the checkbox is selected navigation component will render markup for example:

`<nav role="navigation" class="cmp-navigation">`
  `<button class="cmp-navigation__toggle" aria-label="${label}" aria-expanded="false"></button>`
  `<ul> ...`
  `</ul>`
`</nav>`

The additional requirement is to provide accessibility with:
- aria-hidden (or hiding the button in another way)
- aria-expanded
- check if aria-pressed should be present

What do you think about these improvements? We are open to contributing.

Regards

**Comment by richardhand** [comment]
@zajcu - my instinct is that we shouldn't couple this action button to the navigation itself, rather it should be a generic toggleable action that lives outside of the component (could for example use the button core component, although the accessibility attributes would need to be managed somehow) and simply attaches some handling to show/hide the navigation (or whichever component needs showing/hiding). This is similar to how the Core Component Library hamburger menu is implemented [0].

[0] http://opensource.adobe.com/aem-core-wcm-components/library.html

**Comment by zajcu** [comment]
@richardhand the hamburger as I see in the Core Component Library is a different component configured on the specific breakpoint. So basically you have 2 components with similar logic (rendering links) but different markup. In result, the cost of the maintenance looks bigger.
Additionally, the authoring is quite longer and harder because you need to remember that on specific breakpoint you have a different component with a different configuration.

I'm fine with special sly include/resource button in the navigation component but I'm not sure about Core Component Library solution.

**Comment by richardhand** [comment]
Hi @zajcu,
 
> So basically you have 2 components with similar logic (rendering links) but different markup. In result, the cost of the maintenance looks bigger.
Yes this isn't ideal, but was due to us not wanting to introduce too many custom components in the component library, to show what is possible with Core Components and CSS alone. I wouldn't expect these two components on different break points to be required in a "real" project.

But the separation idea is there - _any_ action button should be able to handle the show hide of the navigation (just a list of links) rather than embedding this functionality into the navigation component, which may or may not be useful depending on the project. It would be the same for the language navigation component (which could have a trigger), a dialog component, and others, so I think a lower maintenance solution is to have a generic show / hide action handler outside of these components.

This could also be handled perhaps in some kind of wrapping header component that adds this action, includes the navigation, search bar, etc and handles the show / hide behaviour.

**Comment by gabrielwalt** [comment]
Example of an accessible navigation menu that show-cases what we should have:
https://pauljadam.com/demos/hamburger-menu.html

**Comment by gabrielwalt** [comment]
Relates to issue #1760

**Comment by gabrielwalt** [comment]
@adobe export issue to Jira project SITES

**Comment by gabrielwalt** [comment]
The [Component Library](https://www.aemcomponents.dev/) show-cases a hamburger menu with the Core navigation component that can be toggled by a Core Button component.

We should improve the Component Library by adding the accessibility attributes using JS.
